pkg/controller/statusmanager/machineconfig_status.go

```diff
@@ -0,0 +1,156 @@
+package statusmanager
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/openshift/cluster-network-operator/pkg/network"
+	"github.com/openshift/cluster-network-operator/pkg/platform"
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (status *StatusManager) SetFromMachineConfigs(ctx context.Context) {
+	status.Lock()
+	defer status.Unlock()
+
+	masterMachineConfigs, err := status.networkMachineConfigs(ctx, platform.MasterRoleMachineConfigLabel)
+	if err != nil {
+		log.Printf("failed to retrieve machine configs for master role: %v", err)
+	}
+	workerMachineConfigs, err := status.networkMachineConfigs(ctx, platform.WorkerRoleMachineConfigLabel)
+	if err != nil {
+		log.Printf("failed to retrieve machine configs for worker role: %v", err)
+	}
+	// When both master and worker roles don't have any machine configs owned by network operator, then return now.
+	if len(masterMachineConfigs) == 0 && len(workerMachineConfigs) == 0 {
+		status.setNotDegraded(MachineConfig)
+		status.unsetProgressing(MachineConfig)
+		return
+	}
```

Comment:
Not sure this is entirely true. Don't we also need to check that the pool status is not referencing any machine config that had been added by us? Otherwise it would mean they are still progressing towards removing our machine configs.